scripts/web_reports.py

```diff
@@ -56,15 +75,67 @@ def web_report_course(summary_path, stat_path, output_path, html_templates, cour
     course_url = "https://www.uio.no/studier/emner/matnat/ifi/"+course_code
 
     main_contents = []
+    additional_js = []
     if language == "NO":
         main_contents.append(r"<h2>Vurdering:</h2>")
-        main_contents.append(r"(Gjennomsnittlige svar på de viktigste spørsmålene)<br />")
+        main_contents.append(r"Velg spørsmål for å se data fra studentenes vurdering:<br />")
     else:
         main_contents.append(r"<h2>Assessment:</h2>")
-        main_contents.append(r"(Average answers to the most important questions)<br />")
-
-    for question, qustion_stats in stats["questions"].items():
-        main_contents.append(r"<i>"+question+"</i> " + qustion_stats["average_text"] + " <br />")
+        main_contents.append(r"Choose a question to see data from the student assessment:<br />")
+
+    options = []
+    questions = []
+    for question, question_stats in stats["questions"].items():
+        question_id = re.sub('[^a-z]+', '', question.lower())
+        chart_id = 'chart_' + question_id
+        questions.append('''
+            <div id="{question_id}" class="question">
+                <h4>{question_label}: {question}</h4>
+                <p>{average_label}: {average}</p>
+                <div id="{chart_id}" class="chart"></div>
+            </div>
+        '''.format(
+            question_id=question_id,
+            question_label='Spørsmål' if language == "NO" else 'Question',
+            question=question,
+            average_label='Gjennsomsnittlig svar' if language == "NO" else 'Average answer',
+            average=question_stats["average_text"],
+            chart_id=chart_id
+        ))
+        additional_js.append(create_chart_js(question, question_stats, scales, chart_id))
+        options.append('<option value="{}">{}</option>'.format(question_id, question))
+
+    main_contents.append('<select id="select_question" onchange="show_selected_question();">')
+    main_contents.extend(options)
+    main_contents.append('</select>')
+    main_contents.append(
+        '<button id="button_show_all_questions" onclick="show_all_questions();">{}</button>'
+            .format('Vis alle' if language == "NO" else 'Show all'))
+    main_contents.extend(questions)
+
+    additional_js.append('''
+        function show_selected_question() {
+            var choice = document.getElementById("select_question").value;
+            var questions = document.querySelectorAll('.question');
+            for (var i = 0; i < questions.length; i++) {
+                questions[i].hidden = true;
+            }
+            document.querySelector('#' + choice).hidden = false;
+        }
+
+        function show_all_questions() {
```

Comment:
Var litt usikker på hvordan jeg skulle gjøre lokaliseringa på det, siden javascripten vel bør ut av denne fila etter hvert. En mulighet er å lage to knapper og vise en av gangen, så lokaliseringa kan være i markup.